### DISCUSSION TITLE
Sidekick and SD2IEC

### DISCUSSION THREAD

**Discussion by gms0012** [frntc/Sidekick64#61] [General]
sometimes when loading the SD card Directory freezes...

when using the "buildin loader" (also external FB16) the system freezes and I need to press "reset" at the C16... after this I see the files in the folder...

using 3B+ :) and Commodore C16

**Comment by frntc** [comment]
You mean SD2IEC, right?

**Comment by gms0012** [comment]
yes

**Reply by frntc** [reply]
ok, you're saying you start FB16 using the Sidekick, and then sometimes you need to reset (and thus launch it again) to make the directory loading work? Sidekick doesn't do much when launching a PRG (such as FB16), and sets the "last used device" to 8. Is your SD2IEC configured as device 8?

**Reply by gms0012** [reply]
yes, my SD2IEC is configured as device 8.

the reason for my question is that I only have the issues when SK64 is connected. When I disconnect it and use SD2IEC I have no problems loading (large) directorys or files...

another example.. i try to use DIRPLUS on SD Card... When i quit to basic and try to load it, it starts to load but then nothing happens.. when I disconnect Sk64 there are no problems and I can use DIRPLUS (or FB16) without any trouble

**Comment by gms0012** [comment]
any ideas what i can do that it works better with SD2IEC or UII+?

**Comment by frntc** [comment]
Sorry, I didn't have time yet to set up my C16 and try to reproduce the problem. I always used a Pi1541 with the C16 and Sidekick without problems. I'm a bit confused now since you mention an U2+?

**Comment by gms0012** [comment]
why confused? U2+ can also be used with C16.... Works like a baby without SK64... same as SD2IEC.

I still have issues with no "loader" connected.. just when I want to load my (large) directory with the build in loader.. it hangs... and I need to reset the SK64 to see the contents of the directory....

i also tried to overclock the Rasp... but no change....

I think u can also speak german...

wenn ich den eingebauten Lader benutze um den Inhalt meiner SD Karte laden will (Verzeichnis PRG264 glaube ich - müsste nachgucken), dann kann ich teilweise durch die Hierarchie durchklicken (Habe eine Ordnerstruktur mit A B C ...".. aber wenn ich dann in einem Verzeichnis mit Dateien bin, hängt sich das "Laden" auf und ich muss einmal RESET drücken. Danach bin ich dann in dem gewählten DIR und sehe auch die Files....

und das ist eigentlich auch so, wenn ich FB16 oder ein anderes Tool nehme... manchmal kommt er über das "Loading" nicht hinaus, zeigt aber nichts an... "Klemme" ich das SK ab funktioniert das SD2IEC oder auch das U2+ wunderbar... ich komme problemlos in dir Directorys und kann auch alles laden...

**Comment by frntc** [comment]
(German is fine for me, but I'll switch to English in case someone has similar issues)

Ah, I've never heard of someone's using the U2+ as standalone with the C16.

But I still wonder, because there are two different things:
- using the Sidekick-Filebrowser runs entirely on the RPi, independent of any IEC devices
- once you exit the Sidekick-Menu you can use the IEC-devices, and Sidekick should leave the computer in a "well-configured way" after loading a PRG. When not loading a PRG it's a plain reset.

How are you powering the Sidekick/RPi? Can you send me a directory listing of your SD-card (mail)?

**Reply by gms0012** [reply]
The U2+ can be used at a C16.. but not as standalone.... I mount files by PC (putty).... it is connected to the "Floppy" port and connected by LAN with my PC - and it also has external power

I power the RPI with an external power supply which is for the RP3...

yes I can send u files / screenshots / videos by mail

"using the Sidekick-Filebrowser runs entirely on the RPi, independent of any IEC devices" and yes.. so there must an "issue" with the SK  somewhere...

**Comment by frntc** [comment]
I suspect the issue then on the SD card :-) It's really just the SK firmware reading the SD card directory contents. A very small chance that some signal is bad (but I'd have to look at my code again). In any case it never hurts to clean the expansion port connectors is something doesn't work as expected ;-)

**Reply by gms0012** [reply]
ok mate...

i will try to delete the PRG files and test again with less files.... first tests will be without any other external device than the SK64 and make sure, that it "loads" correct the folders...

and I will clean the contacts again.
